core/model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java

```diff
@@ -511,7 +506,7 @@ private ModelIterator matchPattern(Resource subj, IRI pred, Value obj, Resource.
 	}
 
 	private Iterator<ModelStatement> find(Statement st) {
-		Resource subj = st.getSubject();
+		SubjectValue subj = st.getSubject();
```

Comment:
This worries me from an end-user perspective. An awful lot of user code will rely on the result of `getSubject()` being a `Resource`. So this will require a lot of refactoring for anyone hoping to bump RDF4J to this version.

Would an alternative be that we get rid of `SubjectValue` altogether, and just make `Triple` extend `Resource`?

Reply:
Triple extends Resource sounds very tempting at first but it will break the model by allowing Triple in the context position, which isn't something RDF* allows. I do realise the proposed changes will break code for some users but the breaking changes are minimal.